nvidia/gpu.go

```diff
@@ -56,6 +57,13 @@ func (g Utilization) run(cmd *exec.Cmd, gpuCount int, query string, action Actio
 	reader := action.start(cmd)
 	gpuIndex := 0
 	events := make([]common.MapStr, gpuCount, 2*gpuCount)
+	bdLink := newBDLink()
+	bdLinkCmd := bdLink.command()
+	links, err := bdLink.run(bdLinkCmd, NewLocal())
```

Comment:
My initial reaction to this sort of thing is that we should be using language-native file ops instead of firing up another shell. Both for performance reasons as well as for robustness.

I'm not sure here, at least about the performance part. I think in go we would use ioutil.ReadDir to get a list of os.FileInfo objects. The FileInfo Mode field can tell us if it's a link. Buuuut to actually get the link target we would have to invoke os.ReadLink on each link...

Did you have thoughts along those lines?

Reply:
I simply replicated the existing functionality in the beat to query for the number of GPUs.
(see original file gpucount.go).
This is my first foray into Go and this seemed a simple approach.
Using the Go provided file ops may be more robust.

Reply:
I rewrote BDLink and implemented a new func getBDDevLinks(linkPath string) (map[int]string, error) that uses Go native file operations to read the sum links.